src/hbbft.erl

```diff
@@ -196,10 +194,8 @@ encrypt(PK, Bin) ->
 
 get_encrypted_key(SK, <<_IV:16/binary, EncKeySize:16/integer-unsigned, EncKey:EncKeySize/binary, _/binary>>) ->
     <<USize:8/integer-unsigned, UBin:USize/binary, V:32/binary, WSize:8/integer-unsigned, WBin:WSize/binary>> = EncKey,
-    %% XXX we don't have a great way to deserialize the elements yet, this is a hack
-    Ugh = tpke_pubkey:hash_message(tpke_privkey:public_key(SK), <<"ugh">>),
-    U = erlang_pbc:binary_to_element(Ugh, UBin),
-    W = erlang_pbc:binary_to_element(Ugh, WBin),
+    U = tpke_pubkey:deserialize_element(tpke_privkey:public_key(SK), UBin),
```

Comment:
Perhaps assign the `tpke_privkey:public_key(SK)` to something like `PubKey` and use that instead of calling it twice.